### DISCUSSION TITLE
"Custom Branding" is not allowing me to customize per station's css. Anyone has it working?

### DISCUSSION THREAD

**Discussion by pmgreb** [AzuraCast/AzuraCast#4701] [General]
> > @rashawntemple The "Custom Branding" page allows you to specify both custom CSS and custom JavaScript to be included on all public-facing pages, including the embed pages. The `<body>` tag on the minimal iframe-style embeds has the class `embed`, so you could just target those pages specifically with:
> > ```css
> > body.embed {
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > in the custom CSS.
> 
> So, what code do use to change the text color of now playing? Asking for noob. lol

What if I want to use this code for a specific radio station instead of the general (all) pages?

_Originally posted by @pmgreb in https://github.com/AzuraCast/AzuraCast/issues/695#issuecomment-946993370_

**Comment by Vaalyn** [comment]
Take a look at the second section of the CSS documentation [`Custom CSS for Public Pages`](https://docs.azuracast.com/en/administration/customization/css#custom-css-for-public-pages) and note the `body.station-my_radio` class.

From the docs page:

> Each station's public pages have a class applied to the <body> element that follows the format station-<station_short_name>.
> 
> So if you have a station called My Radio, this class would be station-my_radio. You can use this to define CSS styles that should only apply to a single station by prefixing your CSS selectors like this:
> ...

This is also the case for the embed pages.

**Reply by pmgreb** [reply]
Can you please send me a working example? because I've done many tests and it's not working...

**Reply by pmgreb** [reply]
> Can you please send me a working example? because I've done many tests and it's not working...

I kept making tests and I got it working 👍 just like:

body.station-radioStationA{

    background: #eeefff url(https://www.azuracast.com/img/bg-masthead.jpg);
    background-size: cover;
    background-attachment: fixed;

}

body.station-radioStationB{

    background: #ffffff url(https://images.unsplash.com/photo-1598743400863-0201c7e1445b?ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8cmFkaW8lMjBicm9hZGNhc3R8ZW58MHx8MHx8&ixlib=rb-1.2.1&w=1000&q=80);
    background-size: cover;
    background-attachment: fixed;

}



Where radioStationA and radioStationB are the name in the url of the radio: https://my-azuracast-server/public/radioStationX

Thanks for your help!!

**Comment by spensyr** [comment]
I currently run 3 streaming stations on a Digital Ocean Droplet, Ubuntu 22.04, Azuracast v0.17.3 Stable • Docker • PHP 8.1

**Please help, anyone, if you have a notion of what I might be doing amiss below.**

When I create the CSS Custom Branding for my station named "CHFXDB", as shown in the graphic below, and go to the CHFXDB Public Page, **the designated graphic file does not show as the BG.**:

![AZ-Custom-Branding-BG-CHFXDB](https://user-images.githubusercontent.com/24966070/195223344-c0f19a7f-b21a-468e-b781-39611c9317b9.png)

I have used all the information I could find here in the Discussions and the Azuracast Docs explaining Custom Branding and CSS excerpts to use for individual station pages. However, the custom background graphic for CHFXDB stubbornly refuses to appear.

FTR, I have flushed the browsers' cache before trying to load the CHFXDB Public Page, and I have attempted to Reload the station Public Page on each use.

**Reply by BusterNeece** [reply]
@spensyr Try prefixing each element the way we do in our example docs: https://docs.azuracast.com/en/administration/customization/css#custom-css-for-public-pages

Our theme system requires this in order to override the theme-specific settings.

**Comment by captainradio5** [comment]
[btw, I am @spensyr, too, I must have two logins and did not realize it]

I tried both [data-theme] and [data-theme="light"] leaders to the CSS shown above with no effect.

However, after I SAVED CHANGES, I immediately went to the Azuracast App Log and found these two lines concurrent with my action:

[2022-10-12T18:42:58.228802+00:00] AzuraCast.NOTICE: SFTP user "test1" not found. [] []
[2022-10-12T18:43:01.890353+00:00] AzuraCast.NOTICE: Starting new Message Queue worker process. {"runtime":0,"workerName":"app_worker_0"} []

I have no clue who "test1" is - "CaptainRadio" is the only DJ at each of my stations.
Otherwise, the CSS is still not working - the Azuracast Unicorn appears.

**Comment by spensyr** [comment]
I also have started getting these errors in my Azuracast General Log:

[2022-10-13T0**2:15:41**.488555+00:00] AzuraCast.ERROR: 
**Request returned status code 400**. {"body":""} {"station":{"id":2,"name":"CHFXDB"}}

[2022-10-13T0**2:15:59**.292563+00:00] AzuraCast.ERROR: 
**Request returned status code 400**. {"body":""} {"station":{"id":2,"name":"CHFXDB"}}

My current Station Profile excerpt screenshot:
![image](https://user-images.githubusercontent.com/24966070/195484658-2959f50b-e78f-4bc7-a50d-ceb9af33c5bb.png)
